common/processes/platform/tiny_process_library/tiny_process_library.cpp

```diff
@@ -123,4 +327,24 @@ void Process::Kill() {
 	}
 }
 
+void ProcessReaderFunctor::operator()(const char *bytes, size_t n) {
+	if (fd_ < 0) {
+		return;
+	}
+
+	size_t written = 0;
+	while (written < n) {
```

Comment:
Q: While is this essentially a retry, instead of an incremental write?

Reply:
I suppose it's both. I don't think `write` is supposed to ever return 0 for a blocking file descriptor. If the pipe is closed I get `errno == EPIPE` instead. So the retry case should in theory never happen, at least it didn't in my testing. But if it did, it should still work.